tests/ui/deref-rc.rs

```diff
@@ -1,8 +0,0 @@
-//@ run-pass
-
-use std::rc::Rc;
-
-fn main() {
-    let x = Rc::new([1, 2, 3, 4]);
-    assert_eq!(*x, [1, 2, 3, 4]);
-}
```

Comment:
Suggestion: I might keep this test for basic coverage, but just move it under idk `tests/ui/deref/` (or similar)

Reply:
This was introduced in https://github.com/rust-lang/rust/pull/14830, and revisiting this, I think deleting it is best.